### PR TITLE
[Importer] Add reshape around softmax

### PR DESCRIPTION
**Pull request by EpureRares**
Summary:
TFLiteImporter: Wrap Softmax by Reshape nodes to force 2D arrays.

Test Plan:
This fix is tested internally.

### CODE REVIEW

**Comment by mciprian13** [comment]
@khabinov Can you review this PR? It fixes a problem for one of the internal models. Thanks!

File: lib/Importer/TFLiteModelLoader.cpp [hunk 1]
```diff
@@ -2119,7 +2119,10 @@ Error TFLiteModelLoader::loadSoftmax(const tflite::Operator *op,
     }
 
     // Create float Softmax regardless of the type defined in the model.
-    output = F_->createSoftMax(opInfo.name, input, selected, nullptr, beta);
+    auto *FN =
+        F_->createFlattenV1("reshapeInput", input, input.dims().size() - 1);
+    auto *SMN = F_->createSoftMax(opInfo.name, FN, selected, nullptr, beta);
+    output = F_->createReshape("reshapeOutput", SMN, outTy->dims());
```
Comment (khabinov) [inline lib/Importer/TFLiteModelLoader.cpp:2125]: Could you please add a comment explaining why such reshapes are needed?

File: lib/Importer/TFLiteModelLoader.cpp [hunk 2]
```diff
@@ -2131,7 +2134,12 @@ Error TFLiteModelLoader::loadSoftmax(const tflite::Operator *op,
       output = F_->createQuantize(opInfo.name + ".Quantize", output, outTy);
     }
   } else {
-    output = F_->createSoftMax(opInfo.name, input, selected, outTy, beta);
+    auto *FN =
+        F_->createFlattenV1("reshapeInput", input, input.dims().size() - 1);
+    TypeRef intermOutTy =
+        F_->getParent()->uniqueTypeWithNewShape(outTy, FN->getResult().dims());
+    auto *SMN = F_->createSoftMax(opInfo.name, FN, selected, intermOutTy, beta);
+    output = F_->createReshape("reshapeOutput", SMN, outTy->dims());
```
Comment (khabinov) [inline lib/Importer/TFLiteModelLoader.cpp:2142]: Same as above.

**Comment by EpureRares** [comment]
@khabinov I added the required comments. Thanks!

**Review by khabinov — COMMENTED** [review]
Looks good, thanks!

**Review by houseroad — APPROVED** [review]

**Pull request closed without merging**